pkgs/servers/mail/dovecot/default.nix

```diff
@@ -1,8 +1,9 @@
-{ stdenv, lib, fetchurl, perl, pkgconfig, systemd, openssl
+{ stdenv, lib, fetchurl, fetchpatch, perl, pkgconfig, systemd, openssl
 , bzip2, zlib, lz4, inotify-tools, pam, libcap
 , clucene_core_2, icu, openldap, libsodium, libstemmer, cyrus_sasl
 , nixosTests
 # Auth modules
+, withLua ? true, lua, autoreconfHook
```

Comment:
I am not sure if we want to enable it by default. Do you think this would be something everyone would need or more of a niche feature?

Reply:
I agree, this must be optional.

Reply:
Why?

Reply:
Because closure size and security

Reply:
This is dovecot. If you don't configure a lua userdb or passdb, this code will never run. Are you running this anywhere?

Reply:
Yes, dovecot is my imap server.  But I see that you have commit access so I assume you're going to merge this anyway.

Reply:
I won't, I've decided that I'm just going to deploy this in my own system and not bother.

Reply:
I don't think that answering a simple question with a thumbs down is how we want to communicate here. If all other distros enable this by default then we can just enable it without any flag. If none of the distros enable it by default then we need to think about it a bit more. If it does not decrease security or increase closure size by that much we can also enable it by default and maybe add a flag to disable it. If it is a feature you definitely want to disable by default then a by default disabled flag is required.

I just think it is a good idea to maybe talk two sentences about such a flag in a program like dovecot which might impact a lot of people before adding a default enabled flag which might or might not cause a big change for a lot of people. If we can't do that then I don't think I work as a reviewer for your PRs.

Reply:
Debian splits Lua authentication into an extra package (https://packages.debian.org/buster/dovecot-auth-lua) and OpenSUSE doesn't seem to have it at all.

Reply:
@SuperSandro2000 "If we can't do that then I don't think I work as a reviewer for your PRs." amen to that. You really don't. Why do you think I blocked you? Because I "value" your "insights". But as I said, I won't push for this anymore. Or much else for that matter, because why would I? So yeah, do all of us a favor and don't touch my PRs anymore.